packages/next/build/swc/src/styled_jsx/utils.rs

```diff
@@ -0,0 +1,368 @@
+
+  (static_class_name, class_name_expr)
+}
+
+pub fn make_external_styled_jsx_el(
+  style: &ExternalStyle,
+  style_import_name: &String,
+) -> JSXElement {
+  let attrs = vec![JSXAttrOrSpread::JSXAttr(JSXAttr {
+    name: JSXAttrName::Ident(Ident {
+      sym: "id".into(),
+      span: DUMMY_SP,
+      optional: false,
+    }),
+    value: Some(JSXAttrValue::JSXExprContainer(JSXExprContainer {
+      expr: JSXExpr::Expr(Box::new(style.expr.clone())),
+      span: DUMMY_SP,
+    })),
+    span: DUMMY_SP,
+  })];
+  let opening = JSXOpeningElement {
+    name: JSXElementName::Ident(Ident {
+      sym: style_import_name.to_string().into(),
+      span: DUMMY_SP,
+      optional: false,
+    }),
+    attrs,
+    span: DUMMY_SP,
+    self_closing: false,
+    type_args: None,
+  };
+
+  let closing = Some(JSXClosingElement {
+    name: JSXElementName::Ident(Ident {
+      sym: style_import_name.to_string().into(),
+      span: DUMMY_SP,
+      optional: false,
+    }),
+    span: DUMMY_SP,
+  });
+
+  let children = vec![JSXElementChild::JSXExprContainer(JSXExprContainer {
+    expr: JSXExpr::Expr(Box::new(Expr::Ident(style.identifier.clone()))),
+    span: DUMMY_SP,
+  })];
+  JSXElement {
+    opening,
+    closing,
+    children,
+    span: DUMMY_SP,
+  }
+}
+
+pub fn make_local_styled_jsx_el(
+  style_info: &LocalStyle,
+  css_expr: Expr,
+  style_import_name: &String,
+) -> JSXElement {
+  let mut attrs = vec![JSXAttrOrSpread::JSXAttr(JSXAttr {
+    name: JSXAttrName::Ident(Ident {
+      sym: "id".into(),
+      span: DUMMY_SP,
+      optional: false,
+    }),
+    value: Some(JSXAttrValue::JSXExprContainer(JSXExprContainer {
+      expr: JSXExpr::Expr(Box::new(string_literal_expr(
+        hash_string(&style_info.hash).clone().as_str(),
+      ))),
+      span: DUMMY_SP,
+    })),
+    span: DUMMY_SP,
+  })];
+
+  if style_info.is_dynamic {
+    attrs.push(JSXAttrOrSpread::JSXAttr(JSXAttr {
+      name: JSXAttrName::Ident(Ident {
+        sym: "dynamic".into(),
+        span: DUMMY_SP,
+        optional: false,
+      }),
+      value: Some(JSXAttrValue::JSXExprContainer(JSXExprContainer {
+        expr: JSXExpr::Expr(Box::new(Expr::Array(ArrayLit {
+          elems: style_info
+            .expressions
+            .iter()
+            .map(|expression| {
+              Some(ExprOrSpread {
+                expr: expression.clone(),
+                spread: None,
+              })
+            })
+            .collect(),
+          span: DUMMY_SP,
+        }))),
+        span: DUMMY_SP,
+      })),
+      span: DUMMY_SP,
+    }));
+  }
+
+  let opening = JSXOpeningElement {
+    name: JSXElementName::Ident(Ident {
+      sym: style_import_name.to_string().into(),
+      span: DUMMY_SP,
+      optional: false,
+    }),
+    attrs,
+    span: DUMMY_SP,
+    self_closing: false,
+    type_args: None,
+  };
+
+  let closing = Some(JSXClosingElement {
+    name: JSXElementName::Ident(Ident {
+      sym: style_import_name.to_string().into(),
```

Comment:
Same here

Reply:
Oh, I found that github erased my previous comments.

`JsWord` can be constructed from `&str`, and if the type is not exactly `&str`, you can do like `(* style_import_name).into()`